packages/material-ui/src/Badge/Badge.js

```diff
@@ -67,7 +51,27 @@ const BadgeRoot = styled(
 const BadgeBadge = styled(
   'span',
   {},
-  { name: 'MuiBadge', slot: 'Badge', overridesResolver },
+  {
+    name: 'MuiBadge',
+    slot: 'Badge',
+    skipVariantsResolver: false,
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+
+      return {
+        ...styles.root,
```

Comment:
Not sure if this is correct, but this is how it was previously implemented, didn't want to change anything, we can revisit later.

Reply:
It doesn't look correct: https://github.com/mui-org/material-ui/blob/1617f4ee72920d40d0b0f9dac34f90b82dadf4bf/packages/material-ui/src/Badge/Badge.js#L194
```suggestion
```